src/api/common_api.js

```diff
@@ -287,6 +321,32 @@ module.exports = {
                                 items: {
                                     type: 'string'
                                 }
+                            },
+                            condition: {
+                                type: 'object',
+                                properties: {
+                                    string_equals: {
+                                        $ref: '#/definitions/bucket_policy_condition'
+                                    },
+                                    string_not_equals: {
+                                        $ref: '#/definitions/bucket_policy_condition'
+                                    },
+                                    string_equals_ignore_case: {
+                                        $ref: '#/definitions/bucket_policy_condition'
+                                    },
+                                    string_not_equals_ignore_case: {
+                                        $ref: '#/definitions/bucket_policy_condition'
+                                    },
+                                    string_like: {
+                                        $ref: '#/definitions/bucket_policy_condition'
+                                    },
+                                    string_not_like: {
+                                        $ref: '#/definitions/bucket_policy_condition'
+                                    },
+                                    null: {
```

Comment:
we probably don't want to use a name such as `null` as a key - this will be error prone in the code.
Maybe call it null_condition or something like that?

Reply:
Null is the key used by AWS. It should stay if we want to store the JSON as is. However, I can change it specifically if it is problematic